grama/tran_pivot.py

```diff
@@ -64,14 +71,163 @@ def tran_pivot_longer (
 
     """
 
+    ########### Pre-Check List #############
+    ### Check if tran_select was used
+    if isinstance(columns, DataFrame):
+        columns = columns.columns.values
+
+    ### Check if names_to is a list or str
+    names_str = False
+    if isinstance(names_to, str):
+        names_str = True
+        if names_sep is not None:
+            raise TypeError("""In order to use names_sep more than 1 value
+                needs to passed to names_to""")
+
+    ### Check for .value input
+    dot_value = False
+    if names_str == False:
+        for i, v in enumerate(names_to):
+            if names_to[i] == ".value":
+                dot_value = True
+    else:
+        if names_to == ".value":
+            dot_value = True
+
+    #######################################
+
+
+    ########### .value pivot #############
+
+    ### Check if .value operation needs to occur
+    if dot_value is True:
+
+        ### collect unused columns to pivot around
+        data_index = collect_indexes(df, columns)
+
+        if names_sep is not None:
+            ### Add index and split column to dataset
+            longer = df.reset_index().melt(
+                    id_vars="index",
+                    var_name="split",
+                    value_vars=columns,
+                    value_name=values_to
+                )
+
+            ### DataFrame Cleanup
+            longer = split_cleanup(
+                longer=longer,
+                names_to=names_to,
+                names_sep=names_sep,
+                values_to=values_to
+            )
+
+        else:
+            ### Add index column and .value column
+            longer = df.reset_index().melt(
+                    id_vars="index",
+                    var_name=".value",
+                    value_vars=columns,
+                    value_name=values_to
+                )
+
+        ### clean up index_to call
+        longer = index_to_cleanup(df, longer, data_index)
+
+        ### arrange what indexes_from should be
+        if names_str is True:
+            indexes = ["index"] + data_index
+        else:
+            names_to = list(names_to)
+            value_loc = names_to.index(".value")
+            if value_loc == 0:
+                indexes = ["index"] + data_index + names_to[1:]
+            else:
+                indexes = ["index"] + data_index + names_to[0:value_loc] \
```

Comment:
Codecov notes that line #L146 is not covered by a test; does this imply functionality that's not covered by the test suite?